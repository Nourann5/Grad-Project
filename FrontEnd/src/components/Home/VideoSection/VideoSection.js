import React from 'react'
import { Col, Container, Row } from 'react-bootstrap'
import styles from './VideoSection.module.css'
import {ReactComponent as CircleIcon} from 'assets/icons/circle.svg'
import {ReactComponent as PlayIcon} from 'assets/icons/play.svg'
import {ReactComponent as JoinPortalIcon} from 'assets/icons/joinPortal.svg'
import videoImgSrc from 'assets/imgs/video.png'
import { Link } from 'react-router-dom'
function VideoSection() {
  return (
    <section id ={styles["video"]}>
        <Container>
            <Row>
                <Col lg='6' xs='12' className="order-lg-1 order-2">
                    <div className="video__content-cont d-flex flex-column">
                        <h2 className={styles["video__heading"]}>A NICE HEADLINE</h2>
                        <p className={styles["video__para"]}>Lorem ipsum dolor sit amet, consetetur sadipsc ingipsum dolor sit amet, consetetur sadipscing</p>
                        <div className={styles["video__features"]}>
                            <div className={styles["video__feature"]}>
                                <CircleIcon className={styles["video__feature-icon"]}/>
                                LOREM IPSAM
                            </div>
                            <div className={styles["video__feature"]}>
                                <CircleIcon className={styles["video__feature-icon"]}/>
                                LOREM IPSAM
                            </div>
                            <div className={styles["video__feature"]}>
                                <CircleIcon className={styles["video__feature-icon"]}/>
                                LOREM IPSAM
                            </div>
                            <div className={styles["video__feature"]}>
                                <CircleIcon className={styles["video__feature-icon"]}/>
                                LOREM IPSAM
                            </div>
                        </div>
                        <Link to='/register' className={styles["video__join-portal"]}>
                            <JoinPortalIcon className="video__join-portal-icon"/>
                            Join Portal to Unlock all the Features
                        </Link>
                    </div>
                </Col>
                <Col lg='6' xs='12' className="order-lg-2 order-1">
                    <div className={styles["video__container"]}>
                        <div className={styles["video__cont"]}>
                            <div className={styles["video__overlay"]}></div>
                            <img src={videoImgSrc} className={styles["video__img"]} alt="video img"/>
                            <button className={styles["video__open-video"]}>
                                <PlayIcon className={styles["video__play-icon"]}/>
                            </button>
                        </div>
                    </div>
                
                </Col>
            </Row>

        </Container>

    </section>
  )
}

export default VideoSection