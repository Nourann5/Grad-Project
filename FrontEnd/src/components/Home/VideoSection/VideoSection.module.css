
#video{
    background-color: white;
    padding: 110px;
}
.video__heading{
    font-size: 40px;
    font-family: 'PlayFairBold';
    margin-bottom: 15px;
}
.video__para{
    font-size: 18px;
    font-family: 'PoppinsMedium';
    margin-bottom: 40px;
}
.video__features{
    display: grid;
    grid-template-columns:repeat(2,1fr);
    margin-bottom: 30px;
}
.video__feature{
    display: flex;
    font-size: 20px;
    font-family: 'PoppinsSemiBold';
    color: black;
    align-items: center;
    margin-bottom: 25px;
}
.video__feature-icon{
    margin-right:18px
}
.video__join-portal{
    background-color: var(--main-color);
    padding: 15px 50px;
    color: white;
    border: 0;
    outline: 0;
    border-radius: 6px;
    transition: var(--transition);
    margin-right: auto;
    display: flex;
    align-items: center;
}
.video__join-portal:hover{
    color: white;
    opacity: .9;
}
.video__join-portal-icon{
    margin-right:10px
}

.video__container{
    position: relative;
    height:100%;
}
.video__open-video{
    --size:100px;
    position: absolute;
    top: calc(50% - var(--size)/2);
    right: calc(50% - var(--size)/2);
    width:var(--size);
    height:var(--size);
    display: flex;
    align-items: center;
    justify-content: center;
    background-color:white;
    border: 0;
    z-index: 10;
    border-radius: 50%;
    cursor: pointer;
}
.video__img{
    position: relative;
    width: 100%;
    height: 100%;
}
.video__overlay{
    position: absolute;
    top: -25px;
    right: -25px;
    background: url('assets/imgs/videoBG.png') repeat-x;
    width: 264px;
    height: 264px;
}
.video__cont{
    position: relative;
}
@media(max-width:1024px){
    #video {
        padding: 110px 40px;
    }
}
@media(max-width:992px){
    #video{
        padding: 70px 20px
    }
    .video__heading{
        margin-top: 30px;
        text-align: center;
    }
    .video__para{
        font-size: 16px;
        text-align: center;
    }
    .video__feature{
        justify-content: center;
    }
    .video__join-portal{
        margin:auto
    }
}
@media(max-width:426px){
    .video__overlay{
        display:none
    }
    .video__feature-icon{
        margin-right: 5px;
    }
    .video__feature{
        font-size: 16px;
    }
    .video__join-portal{
        padding: 10px 20px;
    }
    .video__join-portal-icon{
        width: 50px;
    }
    .video__feature-icon{
        width: 20px;
    }
}