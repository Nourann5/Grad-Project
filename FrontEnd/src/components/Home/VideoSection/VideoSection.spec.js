import {render , screen} from '@testing-library/react'
import { BrowserRouter, Route } from 'react-router-dom'
import VideoSection from './VideoSection'

describe('Test the Home Videos', () => { 
    it('Heading exists',()=>{
        const tree =  render(
            <BrowserRouter>
                <VideoSection/>
            </BrowserRouter>
        )
        const headerTitle = screen.getByRole('heading')
        expect(headerTitle).toBeInTheDocument()
        expect(tree).toMatchSnapshot()
    })
})