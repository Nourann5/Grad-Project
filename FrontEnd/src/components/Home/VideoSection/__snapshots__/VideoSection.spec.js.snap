// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Home Videos Heading exists 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        id="video"
      >
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="order-lg-1 order-2 col-lg-6 col-12"
            >
              <div
                class="video__content-cont d-flex flex-column"
              >
                <h2
                  class="video__heading"
                >
                  A NICE HEADLINE
                </h2>
                <p
                  class="video__para"
                >
                  Lorem ipsum dolor sit amet, consetetur sadipsc ingipsum dolor sit amet, consetetur sadipscing
                </p>
                <div
                  class="video__features"
                >
                  <div
                    class="video__feature"
                  >
                    <svg
                      class="video__feature-icon"
                    >
                      circle.svg
                    </svg>
                    LOREM IPSAM
                  </div>
                  <div
                    class="video__feature"
                  >
                    <svg
                      class="video__feature-icon"
                    >
                      circle.svg
                    </svg>
                    LOREM IPSAM
                  </div>
                  <div
                    class="video__feature"
                  >
                    <svg
                      class="video__feature-icon"
                    >
                      circle.svg
                    </svg>
                    LOREM IPSAM
                  </div>
                  <div
                    class="video__feature"
                  >
                    <svg
                      class="video__feature-icon"
                    >
                      circle.svg
                    </svg>
                    LOREM IPSAM
                  </div>
                </div>
                <a
                  class="video__join-portal"
                  href="/register"
                >
                  <svg
                    class="video__join-portal-icon"
                  >
                    joinPortal.svg
                  </svg>
                  Join Portal to Unlock all the Features
                </a>
              </div>
            </div>
            <div
              class="order-lg-2 order-1 col-lg-6 col-12"
            >
              <div
                class="video__container"
              >
                <div
                  class="video__cont"
                >
                  <div
                    class="video__overlay"
                  />
                  <img
                    alt="video img"
                    class="video__img"
                    src="video.png"
                  />
                  <button
                    class="video__open-video"
                  >
                    <svg
                      class="video__play-icon"
                    >
                      play.svg
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      id="video"
    >
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="order-lg-1 order-2 col-lg-6 col-12"
          >
            <div
              class="video__content-cont d-flex flex-column"
            >
              <h2
                class="video__heading"
              >
                A NICE HEADLINE
              </h2>
              <p
                class="video__para"
              >
                Lorem ipsum dolor sit amet, consetetur sadipsc ingipsum dolor sit amet, consetetur sadipscing
              </p>
              <div
                class="video__features"
              >
                <div
                  class="video__feature"
                >
                  <svg
                    class="video__feature-icon"
                  >
                    circle.svg
                  </svg>
                  LOREM IPSAM
                </div>
                <div
                  class="video__feature"
                >
                  <svg
                    class="video__feature-icon"
                  >
                    circle.svg
                  </svg>
                  LOREM IPSAM
                </div>
                <div
                  class="video__feature"
                >
                  <svg
                    class="video__feature-icon"
                  >
                    circle.svg
                  </svg>
                  LOREM IPSAM
                </div>
                <div
                  class="video__feature"
                >
                  <svg
                    class="video__feature-icon"
                  >
                    circle.svg
                  </svg>
                  LOREM IPSAM
                </div>
              </div>
              <a
                class="video__join-portal"
                href="/register"
              >
                <svg
                  class="video__join-portal-icon"
                >
                  joinPortal.svg
                </svg>
                Join Portal to Unlock all the Features
              </a>
            </div>
          </div>
          <div
            class="order-lg-2 order-1 col-lg-6 col-12"
          >
            <div
              class="video__container"
            >
              <div
                class="video__cont"
              >
                <div
                  class="video__overlay"
                />
                <img
                  alt="video img"
                  class="video__img"
                  src="video.png"
                />
                <button
                  class="video__open-video"
                >
                  <svg
                    class="video__play-icon"
                  >
                    play.svg
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
